### PR TITLE
CB-20281 improve caching of cost calculation logic

### PR DESCRIPTION
**Pull request by horadla23**
See detailed description in the commit message.

### CODE REVIEW

File: cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/cost/AwsPricingCache.java [hunk 1]
```diff
@@ -83,44 +65,39 @@ public double getPriceForInstanceType(String region, String instanceType) {
...
     }
 
+    @Cacheable(cacheNames = "awsCostCache", key = "{ #region, #instanceType }")
     public int getMemoryForInstanceType(String region, String instanceType, ExtendedCloudCredential extendedCloudCredential) {
         PriceListElement priceListElement = getPriceList(region, instanceType);
         String memory = priceListElement.getProduct().getAttributes().getMemory();
         return Integer.parseInt(memory.replaceAll("\\D", ""));
     }
 
+    @Cacheable(cacheNames = "azureCostCache", key = "{ #region, #storageType, #volumeSize }")
```
Comment (szabolcs-horvath) [inline cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/cost/AwsPricingCache.java:81]: This should be `"awsCostCache"`, I assume.

Comment (horadla23) [reply cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/cost/AwsPricingCache.java]: yes, thx!

**Review by szabolcs-horvath — APPROVED** [review]

File: cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/cost/AwsPricingCache.java [hunk 2]
```diff
@@ -83,44 +65,39 @@ public double getPriceForInstanceType(String region, String instanceType) {
...
-            LOGGER.info("PriceList for region [{}] and instance type [{}] found in cache.", region, instanceType);
-            return value;
+        GetProductsResult productsResult = getProducts(region, instanceType);
+        String priceListString = productsResult.getPriceList().stream().findFirst()
+                .orElseThrow(() ->
+                        new NotFoundException(String.format("Couldn't find the price list for the region %s and instance type %s!", region, instanceType)));
+        try {
+            return JsonUtil.readValue(priceListString, PriceListElement.class);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```
Comment (lacikaaa) [inline cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/cost/AwsPricingCache.java:100]: could you replace this with a meaningful excpetion?

Comment (horadla23) [reply cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/cost/AwsPricingCache.java:100]: is it OK if I'll do it in another PR? this is not a new code, only if condition was removed

Comment (horadla23) [reply cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/cost/AwsPricingCache.java:100]: nvm, i'll fix it here

**Pull request merged**